### DISCUSSION TITLE
Is it possible to draw a vertical line on a line chart? (Not as a decoration.)

### DISCUSSION THREAD

**Discussion by SSHshadow222** [patrykandpatrick/vico#779] [Questions]
### Question

By default, adding different Y values for the same X value only draws the maximum Y value on a line chart. To illustrate this, I've created a point for each rendered value, and here's the result when running the transaction `lineSeries { series(listOf(1, 1, 1), listOf(1, 2, 3)) }`:

![Screenshot_20240701-110108~2](https://github.com/patrykandpatrick/vico/assets/113729758/34126c09-ea33-43e0-b863-45e44d20620d)

However, my desired result would look more like this:

![oj](https://github.com/patrykandpatrick/vico/assets/113729758/790b428b-4532-4b38-9df7-a0efdccba734)

Is there a way to achieve this?

### Vico version(s)

2.0.0-alpha.16

### UI framework(s)

Jetpack Compose

**Comment by SSHshadow222** [comment]
Here's the code that produced the result visible in the question's first image:

```
val modelProducer = remember { CartesianChartModelProducer.build() }

CartesianChartHost(
    chart = rememberCartesianChart(
        rememberLineCartesianLayer(
            vicoTheme.lineCartesianLayerColors.map {
                rememberLineSpec(
                    backgroundShader = null,
                    point = rememberShapeComponent(Shape.Pill),
                )
            }
        ),
        startAxis = rememberStartAxis(),
        bottomAxis = rememberBottomAxis()
    ),
    modelProducer = modelProducer
)

LaunchedEffect(Unit) {
    modelProducer.tryRunTransaction {
        lineSeries {
            series(
                x = listOf(1, 1, 1),
                y = listOf(1, 2, 3)
            )
        }
    }
}
```

**Comment by Gowsky** [comment]
Hi @SSHshadow222, no, it's not possible currently. May I ask what your use case is for having three y values associated with a single x value?

**Comment by cathua** [comment]
@Gowsky I am actually running into a similar issue. I am creating a chart to depict someone's measurements (for example, weight), where I display a list of readings over time.

If someone has multiple readings in a day, I would like to connect them vertically in a similar way.

**Reply by Gowsky** [reply]
Why would that be desired? The line in a line chart is supposed to connect consecutive entries such that, by following it, you go through them in order. This helps understand how the value changes over time. In your case, regardless of the entries’ order, the user would see one vertical line connecting all of them, which is meaningless. Additionally, since it’s not guaranteed that the oldest entry has the lowest value and the  newest entry has the highest value, you could end up with something like this:

![Frame_6.png](https://github.com/user-attachments/assets/4dcf35d8-bc82-4e5a-b692-cb55a624a2c7)

Both of these things go against how a standard line chart works. It would seem more appropriate to position the points according to their full timestamps, so they can’t share the same _x_-values, while keeping one _x_-axis label per day.

**Reply by cathua** [reply]
I see what you mean. It looks like in my case, it would be better to add some kind of custom decorator to my chart to highlight the values that have the same x-value.

In that case, it looks like I would have to create a custom decorator (similar to the HorizonalLine decorator) that is like HorizontalLine or HorizontalBox but with a specific y range. Is that something that's feasible?

here is a screenshot of what I am trying to achieve:
<img width="167" height="266" alt="Screenshot 2025-08-04 at 18 06 10" src="https://github.com/user-attachments/assets/0bd2ea77-9a65-4981-9a56-653eefa81dab" />

**Reply by Gowsky** [reply]
Hello, @cathua. Yes, this is feasible.

**Reply by cathua** [reply]
@Gowsky could you point me to some documentation to showcase how I could make this custom decoration?

**Reply by Gowsky** [reply]
Please see the API reference and the implementations of the built-in decorations. They mostly use `LineComponent` and `ShapeComponent` to draw the content onto the generic `android.graphics.Canvas`.
